### PR TITLE
:bug: Write YAML stream with correct start marker

### PR DESCRIPTION
**Pull request by mcristina422**
I came across an issue reading `role.yaml` using a package in k8s.io/apimachinery/pkg/util/yaml. The blank new line at the beginning of the file made the decoder error. I _think_ this violates the YAML spec, according to https://yaml.org/spec/1.2/spec.html#id2800132, but I'm definitely not an expert. 

This changes the YAML generator to separate objects simply with `---\n`. 

I was curious about how the internals of https://github.com/kubernetes-sigs/kustomize printed mutli object yaml, but I didn't dig deep enough to find the implementation.

### CODE REVIEW

File: pkg/genall/genall.go [hunk 1]
```diff
@@ -103,7 +103,7 @@ func (g GenerationContext) WriteYAML(itemPath string, objs ...interface{}) error
 		if err != nil {
 			return err
 		}
-		n, err := out.Write(append([]byte("\n---\n"), yamlContent...))
+		n, err := out.Write(append([]byte("---\n"), yamlContent...))
```
Comment (DirectXMan12) [inline pkg/genall/genall.go]: is `yaml.Marshal` guaranteed to produce a newline at the end?

Comment (DirectXMan12) [reply pkg/genall/genall.go]: (i.e. is this safe to not have the extra newline?)

Comment (DirectXMan12) [reply pkg/genall/genall.go]: I think we actually have a helper burried in the k8s libraries for this

Comment (DirectXMan12) [reply pkg/genall/genall.go]: (nope, just a decoder)

Comment (mcristina422) [reply pkg/genall/genall.go:106]: I think `yaml.Marshal` is guaranteed to produce a newline

**Pull request closed without merging**